### PR TITLE
Etrog

### PR DESCRIPTION
**Pull request by revitteth**
- Etrog compatibility changes
- Configurable l1 topic hashes for sequence/validate
- Configurable contracts for PolygonRollupManager and Rollup
- Etrog devnet network config

### CODE REVIEW

File: cmd/cdk-erigon/__debug_bin1256010388
Comment (revitteth) [inline cmd/cdk-erigon/__debug_bin1256010388:1]: Let's remove

File: core/blockchain.go [hunk 1]
```diff
@@ -162,11 +162,11 @@ func ExecuteBlockEphemerally(
 			}
 		}
 
-		// [zkevm] - set smt root hash in magic account
-		err = ibs.ScalableSetSmtRootHash(roHermezDb)
-		if err != nil {
-			return nil, err
-		}
+		// // [zkevm] - set smt root hash in magic account
```
Comment (revitteth) [inline core/blockchain.go]: Can probably delete

File: core/vm/contracts.go [hunk 2]
```diff
@@ -110,11 +110,28 @@ var PrecompiledContractsBLS = map[libcommon.Address]PrecompiledContract{
 	libcommon.BytesToAddress([]byte{18}): &bls12381MapG2{},
 }
 
+var PrecompiledContractsEtrog = map[libcommon.Address]PrecompiledContract{
```
Comment (revitteth) [inline core/vm/contracts.go]: Move to contracts_zkevm

File: hermezconfig-etrog.yaml.example [hunk 3]
```diff
@@ -0,0 +1,19 @@
+datadir : '/Volumes/External/hermez/etrog'
+chain : "hermez-etrog"
+http : true
+private.api.addr : "localhost:9093"
+zkevm.l2-chain-id: 10010
+zkevm.l2-sequencer-rpc-url: "http://zkevm-json-rpc:8123"
+zkevm.l2-datastreamer-url: "34.175.178.16:6900"
+zkevm.l1-chain-id: 11155111
+zkevm.l1-rpc-url: "https://rpc.eu-central-1.gateway.fm/v4/ethereum/non-archival/sepolia?apikey=IkrTRMPAlsjfgTyKz1Xfa5ldS8XSc0PN.En5aU2Gwdxvf3zk2"
```
Comment (revitteth) [inline hermezconfig-etrog.yaml.example:9]: Needs new config values (rollup, rollupmanager etc.)

**Review by V-Staykov — APPROVED** [review]

**Pull request merged**